### DISCUSSION TITLE
Update Color Data object to include HSL value

### DISCUSSION THREAD

**Discussion by tomkiss** [craftcms/cms#10235] [Ideas] [answered]
[HSL color values](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/hsl()) make it much easier to adjust lightness and saturation, which is especially useful for theming. Since I switched most of my CSS color values to HSL a few years ago, it's made working with colour much easier.

It would be helpful if the Color Data object in Craft could return an HSL value along with its existing RGB values.

It's worth noting that there are plugins currently available that can achieve this, but as a relatively small piece of functionality I think it could be of benefit to include this conversion in the core Color Data object functionality.

I just noticed that the UI for the Color Field includes HSL, which I think makes it more a bit more compelling as a feature request.

**Comment by brandonkelly** [comment, marked as answer]
Added for the next release! (6c359ab406a26e432d6912c44f4f246491877dc2)

**Reply by tomkiss** [reply]
Amazing, thanks Brandon. Quick wins! 🙌

**Reply by brandonkelly** [reply]
Craft 3.7.26 is out now with the new methods ✨
